crates/environ/src/vmoffsets.rs

```diff
@@ -152,6 +164,7 @@ impl<P: PtrSize> VMOffsets<P> {
             num_defined_functions: cast_to_u32(module.functions.len()),
             num_defined_tables: cast_to_u32(module.table_plans.len()),
             num_defined_memories: cast_to_u32(module.memory_plans.len()),
+            num_owned_memories: cast_to_u32(module.memory_plans.len() - num_shared_memories),
```

Comment:
Similar to above, I think that the calculation here needs to exclude imported memories first, and then exclude shared memories from the remaining set of memories to figure out the number of owned memories.